broker/lib/controllers/ServiceFabrikApiController.js

```diff
@@ -254,6 +338,97 @@ class ServiceFabrikApiController extends FabrikBaseController {
       });
   }
 
+  startBackup_sf20(req, res) {
+    let backup_started_at;
+    req.manager.verifyFeatureSupport(CONST.OPERATION_TYPE.BACKUP);
+    const trigger = _.get(req.body, 'trigger', CONST.BACKUP.TRIGGER.ON_DEMAND);
+    let backup_guid, deploymentName;
+    return Promise
+      .try(() => this.checkQuota(req, trigger))
+      .then(() => Promise.all([utils
+        .uuidV4(),
+        req.manager
+        .findNetworkSegmentIndex(req.params.instance_id)
+        .then(networkIndex => req.manager.getDeploymentName(req.params.instance_id, networkIndex))
+      ]))
+      .spread((guid, deployment) => {
+        _.set(req.body, 'trigger', trigger);
+        backup_guid = guid;
+        const backup_options = {
+          guid: backup_guid,
+          deployment: deployment,
+          instance_guid: req.params.instance_id,
+          plan_id: req.body.plan_id,
+          service_id: req.body.service_id,
+          context: req.body.context
+        };
+        // Acquire read lock for resource resourceId
+        logger.info(`Attempting to acquire lock on deployment with instanceid: ${req.params.instance_id} `);
+        return lockManager.lock(req.params.instance_id, {
+            lockType: CONST.ETCD.LOCK_TYPE.READ,
+            lockedResourceDetails: {
+              resourceType: CONST.RESOURCE_TYPES.BACKUP,
+              resourceName: CONST.RESOURCE_NAMES.DEFAULT_BACKUP,
+              resourceId: backup_guid
+            }
+          })
+          .then(() => {
+            return eventmesh.server.annotateResource({
+              resourceId: req.params.instance_id,
+              annotationName: CONST.OPERATION_TYPE.BACKUP,
+              annotationType: CONST.APISERVER.RESOURCE_NAMES.DEFAULT_BACKUP,
+              annotationId: backup_guid,
+              val: JSON.stringify(backup_options)
+            });
+          });
+      })
+      .then(() => {
+        backup_started_at = new Date()
+        //check if resource exist, else create and then update
+        return Promise.try(() => eventmesh.server.getResource('deployment', 'directors', req.params.instance_id))
+          .catch(() => eventmesh.server.createResource(null, req.params.instance_id, {}))
+          .then(() => eventmesh.server.updateLastAnnotation({
+            resourceId: req.params.instance_id,
+            annotationName: CONST.OPERATION_TYPE.BACKUP,
+            annotationType: CONST.APISERVER.RESOURCE_NAMES.DEFAULT_BACKUP,
+            value: backup_guid
+          }))
+          .then(() => ServiceFabrikApiController.getResourceAnnotationStatus({
+            resourceId: req.params.instance_id,
+            annotationId: backup_guid,
+            start_state: CONST.RESOURCE_STATE.IN_QUEUE,
+            started_at: backup_started_at
+          }));
+      })
+      .tap(response => {
+        logger.info(`Backup Response `, response);
+        const directorManager = req.manager;
+        return directorManager
```

Comment:
For 2.0, I dont think we need to send the deployment guid calling `directorManager`. Earlier, this was to update the cf status. I think only sending that the backup is triggered for a certain instance_id is enough. From sf2.0, we should avoid calling directormanager.

Reply:
Removing this tap as it is non functional